localization/util/pose_initializer/src/pose_initializer_core.cpp

```diff
@@ -187,27 +176,33 @@ bool PoseInitializer::getHeight(
   return true;
 }
 
-bool PoseInitializer::callAlignService(
-  const geometry_msgs::PoseWithCovarianceStamped & input_pose_msg,
-  const geometry_msgs::PoseWithCovarianceStamped::Ptr & output_pose_msg_ptr)
+void PoseInitializer::callAlignServiceAndPublishResult(
+  const geometry_msgs::msg::PoseWithCovarianceStamped::SharedPtr input_pose_msg)
 {
-  autoware_localization_srvs::PoseWithCovarianceStamped srv;
-  srv.request.pose_with_cov = input_pose_msg;
-
-  ROS_INFO("[pose_initializer] call NDT Align Server");
-  if (ndt_client_.call(srv)) {
-    ROS_INFO("[pose_initializer] called NDT Align Server");
-    // NOTE temporary cov
-    srv.response.pose_with_cov.pose.covariance[0] = 1.0;
-    srv.response.pose_with_cov.pose.covariance[1 * 6 + 1] = 1.0;
-    srv.response.pose_with_cov.pose.covariance[2 * 6 + 2] = 0.01;
-    srv.response.pose_with_cov.pose.covariance[3 * 6 + 3] = 0.01;
-    srv.response.pose_with_cov.pose.covariance[4 * 6 + 4] = 0.01;
-    srv.response.pose_with_cov.pose.covariance[5 * 6 + 5] = 0.2;
-    *output_pose_msg_ptr = srv.response.pose_with_cov;
-    return true;
-  } else {
-    ROS_ERROR("[pose_initializer] could not call NDT Align Server");
-    return false;
+  if (request_id_ != response_id_) {
+    throw std::runtime_error("Did not receive response for previous NDT Align Server call");
```

Comment:
Throwing an error and terminating the program might be too conservative.
We can modify this so that it only outputs RCLCPP_ERROR message and returns.
That way, users can just try initializing again after waiting for service call to finish without restarting the node.

Reply:
```suggestion
    RCLCPP_ERROR(get_logger(), "Did not receive response for previous NDT Align Server call");
    return;
```

Reply:
Sure, done.